api/envoy/admin/v3/config_dump.proto

```diff
@@ -381,3 +381,19 @@ message EndpointsConfigDump {
   // The dynamically loaded endpoint configs.
   repeated DynamicEndpointConfig dynamic_endpoint_configs = 3;
 }
+
+// Message of unready targets information of listener's init manager.
+message ListenerUnreadyTargetsConfigDump {
```

Comment:
Great idea! Thanks for the review. Updated

Reply:
However, for the second part, I think include name into lambda function seems not feasible, since the internal watcher of init manager is defined at Ctor of ManagerImpl and could not pass a string name there. 

What about another solution: since now I have changed the internal watcher of init manager to be fixed to the Ctor with ```std::function<void(string)>```, and for the feature of init manager 'check unready targets', we have to make that ```watcher_``` to be constructed with that. Is it possible that I just split them into two types of watchers. One for internal watcher of init manager(which must be ```<void(string)>```, and the other one type of watcher for non-initmanager, which mostly is for target.

Reply:
I think you can do the other way at least? i.e. wrapping ReadyFn into ReadyFnSendName (I don't think this type is good btw).

i.e.
```
ReadyFn fn;
ReadyFnSendName wrapped = [fn](absl::string_view) -> void { fn(); };
```

Reply:
Yes, I get the idea of wrapping fn into lambda. But the problem is, if I want to return the wrapped function, then watcher will become ```watcher_(name, [this] () { wrapped; return wrapped; })```. I will need to change ```ReadyFn=std::function<void()>``` into ```ReadyFn=std::function<std::function<void(string)>()>```.

Reply:
```C++
using ReadyFnSendName = std::function<void(absl::string_view)>;
using ReadyFn = std::function<ReadyFnSendName()>;

watcher_(name_, [this]() {
  ReadyFnSendName fn = [this](absl::string_view name) {
    onTargetReadySendTargetName(name);
  };
  return fn;
}) {}


auto locked_fn(fn_.lock());
if (locked_fn) {
  // If we can "lock" a shared pointer to the watcher's callback function, call it.
  ENVOY_LOG(debug, "{} initialized, notifying {}", handle_name_, name_);
  auto send_name_fn = (*locked_fn)();
  send_name_fn(handle_name_);
  // (*locked_fn)();
  return true;
}
```

Reply:
This is what wrapping Fn looks like. Since ```watcher_```'s function can not contain any parameter, we have to change its return type from ```void``` to ```ReadyFnSendName```.

Reply:
> I think you can do the other way at least? i.e. wrapping ReadyFn into ReadyFnSendName (I don't think this type name is good btw).
> 
> i.e.
> 
> ```
> ReadyFn fn;
> ReadyFnSendName wrapped = [fn](absl::string_view) -> void { fn(); };
> ```

The wrapping seems to not solve this problem, because the ```onTargetReady()``` now receives a string parameter. And the major part is inside ```fn```, which is passed from manager. The wrapping still could not change the truth the  ```fn``` needs to receive string.

Reply:
> I think you can do the other way at least? i.e. wrapping ReadyFn into ReadyFnSendName (I don't think this type name is good btw).
> i.e.
> 
> ```
> ReadyFn fn;
> ReadyFnSendName wrapped = [fn](absl::string_view) -> void { fn(); };
> ```

Now that the callback function ```onTargetReady```  will receive a string parameter, please tell me how can I wrap this function a ReadyFn then call it while passing a string parameter.